### DISCUSSION TITLE
[gcc] install failes to apply "bigsur" patches

### DISCUSSION THREAD

**Discussion by jacktose** [Homebrew/discussions#288] [Linux] [answered]
`brew install gcc` fails to apply some patches. This is a fresh linuxbrew install on CentOS. (I don't have sudo, hence building from source and the second-choice prefix.)

```
$ brew install gcc
Warning: Building gcc from source as the bottle needs:
- HOMEBREW_CELLAR: any (yours is /data/shared/lusers/jacktose/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /data/shared/lusers/jacktose/.linuxbrew)
- HOMEBREW_REPOSITORY: /home/linuxbrew/.linuxbrew/Homebrew (yours is /data/shared/lusers/jacktose/.linuxbrew/Homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/7baf6e2f/gcc/bigsur.diff
Already downloaded: /usr/lusers/jacktose/.cache/Homebrew/downloads/b09b60b5cbfe9a4f8def153bcb9a44251d54a2caa572c63ca5d9ffab785637ce--bigsur.diff
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-5.5.0/gcc-5.5.0.tar.xz
Already downloaded: /usr/lusers/jacktose/.cache/Homebrew/downloads/62d8cc20018aa926380222fa7098af4384ca68c3ebdabe0806f2cd801ea0d075--gcc-5.5.0.tar.xz
==> Patching
==> Applying bigsur.diff
patching file gcc/config/darwin-c.c
Hunk #1 FAILED at 692.
1 out of 1 hunk FAILED -- saving rejects to file gcc/config/darwin-c.c.rej
patching file gcc/config/darwin-driver.c
Hunk #1 FAILED at 65.
Hunk #2 FAILED at 104.
Hunk #3 succeeded at 29 (offset -86 lines).
Hunk #4 succeeded at 47 with fuzz 1 (offset -84 lines).
Hunk #5 FAILED at 148.
3 out of 5 hunks FAILED -- saving rejects to file gcc/config/darwin-driver.c.rej

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
gcc: align with upstream https://github.com/Homebrew/linuxbrew-core/pull/21380
```

[#21380] doesn't seem directly relevant, but maybe it'll moot this issue? Doesn't help much today.

[Here's the rest of the usual info.][gist] Note that:
The deps are installed fine (from source).
The install somehow didn't create logs (I couldn't find them manually either (despite the error messages above)).
I don't control this system, so that's about as clean as `brew doctor` is gonna get.

I know this is only a semi-supported configuration, but I'd be grateful for any help pointing me in the right direction. Even just why it's applying “bigsur” & “darwin” patches on Linux???

[#21380]: https://github.com/Homebrew/linuxbrew-core/pull/21380
[gist]: https://gist.github.com/jacktose/f123aea34c3d0596f26966b5dfb4eb18

**Comment by carlocab** [comment]
I think you can open a PR to fix this. Here's the code that I think might need to only apply on Macs:

https://github.com/Homebrew/linuxbrew-core/blob/43abb8cc946c93aaf4dbd2aba12ba0612bef8059/Formula/gcc.rb#L58-L63

**Reply by jacktose** [reply]
Thanks. Looks like someone smarter got there first.

**Comment by jacktose** [comment, marked as answer]
Already fixed.
